data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -0,0 +1,296 @@
+                LambdaCommonConfig.REQUEST_RESPONSE, REQUEST_RESPONSE_LAMBDA
+        );
+        invocationType = invocationTypeMap.get(lambdaProcessorConfig.getInvocationType());
+
+        bufferedEventHandles = new LinkedList<>();
+        events = new ArrayList();
+
+        lambdaClient = LambdaClientFactory.createLambdaClient(lambdaProcessorConfig.getAwsAuthenticationOptions(),
+                lambdaProcessorConfig.getMaxConnectionRetries()
+                , awsCredentialsSupplier);
+
+        this.bufferFactory = new InMemoryBufferFactory();
+        try {
+            currentBuffer = this.bufferFactory.getBuffer(lambdaClient, functionName, invocationType);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
+        if (records.isEmpty()) {
+            return records;
+        }
+
+        //lambda mutates event
+        List<Record<Event>> resultRecords = new ArrayList<>();
+
+        for (Record<Event> record : records) {
+            final Event event = record.getData();
+
+            if (whenCondition != null && !expressionEvaluator.evaluateConditional(whenCondition, event)) {
+                continue;
+            }
+
+            try {
+                if (currentBuffer.getEventCount() == 0) {
+                    codec.start(currentBuffer.getOutputStream(), event, codecContext);
+                }
+                codec.writeEvent(event, currentBuffer.getOutputStream());
+                int count = currentBuffer.getEventCount() + 1;
+                currentBuffer.setEventCount(count);
+
+                // flush to lambda and update result record
+                flushToLambdaIfNeeded(resultRecords);
+            } catch (Exception e) {
+                numberOfRecordsFailedCounter.increment(currentBuffer.getEventCount());
+                LOG.error(EVENT, "There was an exception while processing Event [{}]" + ", number of events dropped={}", event, e, numberOfRecordsFailedCounter);
+                //reset buffer
+                try {
+                    currentBuffer = bufferFactory.getBuffer(lambdaClient, functionName, invocationType);
+                } catch (IOException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }
+        }
+        return resultRecords;
+    }
+
+    @Override
+    public void prepareForShutdown() {
+
+    }
+
+    @Override
+    public boolean isReadyForShutdown() {
+        return false;
+    }
+
+    @Override
+    public void shutdown() {
+
+    }
+
+    void flushToLambdaIfNeeded(List<Record<Event>> resultRecords) throws InterruptedException, IOException {
+
+        LOG.info("Flush to Lambda check: currentBuffer.size={}, currentBuffer.events={}, currentBuffer.duration={}", currentBuffer.getSize(), currentBuffer.getEventCount(), currentBuffer.getDuration());
```

Comment:
Is there excessive logging in this method ?

Reply:
sure will reduce them.